### PR TITLE
Adding example for using MapStruct with field mappings

### PR DESCRIPTION
**Pull request by filiphr**

### CODE REVIEW

File: mapstruct-field-mapping/src/main/java/org/mapstruct/example/mapper/OrderItemMapper.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * @author Filip Hrisafov
+ */
+@Mapper
+public interface OrderItemMapper {
+
+    OrderItemMapper MAPPER = Mappers.getMapper(OrderItemMapper.class);
+
+
```
Comment (sjaakd) [inline mapstruct-field-mapping/src/main/java/org/mapstruct/example/mapper/OrderItemMapper.java]: 2 enters

File: mapstruct-field-mapping/src/test/java/org/mapstruct/example/CustomerMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        OrderItem order1 = new OrderItem();
+        order1.setName( "Table" );
+        order1.setQuantity( 2L );
+        customer.setOrderItems( Collections.singleton( order1 ) );
+
+        CustomerDto customerDto = CustomerMapper.MAPPER.fromCustomer( customer );
+
+        assertThat( customerDto.id ).isEqualTo( 10 );
+        assertThat( customerDto.customerName ).isEqualTo( "Filip" );
+        assertThat( customerDto.orders )
```
Comment (sjaakd) [inline mapstruct-field-mapping/src/test/java/org/mapstruct/example/CustomerMapperTest.java]: although I like java8.... we don't rely  on it yet in our standard code..

**Comment by sjaakd** [comment]
2 minor comments. When fixed.. please squash and merge. Very nice 👍

File: mapstruct-field-mapping/src/test/java/org/mapstruct/example/CustomerMapperTest.java (hunk 2 above)
Comment (filiphr) [inline mapstruct-field-mapping/src/test/java/org/mapstruct/example/CustomerMapperTest.java]: You probably meant about the extracting assertions and not this line. In any case I changed it by using the name of the properties in the extraction. Is that fine with you?

Comment (sjaakd) [inline mapstruct-field-mapping/src/test/java/org/mapstruct/example/CustomerMapperTest.java]: I meant the Lamda expression.. Could be on the next line. Sorry.

**Comment by sjaakd** [comment]
Ok.. please go ahead and merge..

**Comment by filiphr** [comment]
Thanks for the review @sjaakd

**Pull request merged**